bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/onvif/OnvifCodec.java

```diff
@@ -50,8 +50,28 @@ public void channelRead(@Nullable ChannelHandlerContext ctx, @Nullable Object ms
         }
         try {
             if (msg instanceof HttpResponse response) {
-                if (response.status().code() != 200) {
-                    logger.trace("ONVIF replied with code {} message is {}", response.status().code(), msg);
+                switch (response.status().code()) {
+                    case 200:
+                        break;
+                    case 401:
+                        if (!response.headers().isEmpty()) {
+                            String authenticate = "";
+                            for (CharSequence name : response.headers().names()) {
+                                for (CharSequence value : response.headers().getAll(name)) {
+                                    if ("WWW-Authenticate".equalsIgnoreCase(name.toString())) {
+                                        authenticate = value.toString();
+                                        logger.debug(
+                                                "ONVIF replied with WWW-Authenticate header:{}, camera may require ONVIF Profile-T support.",
+                                                authenticate);
```

Comment:
It is a detail but you could simply declare the variable inside the if as it is only used for the log:
String authenticate = value.toString();

Reply:
Or even better, not declare at all this variable ;)
And just use `value.toString()` as parameter of logger.debug

Reply:
Fixed and thank you for reviewing it.